File: src/main/java/ru/ioleksiv/telegram/bot/core/controller/annotations/parser/unpacker/PreCheckoutQueryUnpackerParser.java
package ru.ioleksiv.telegram.bot.core.controller.annotations.parser.unpacker;

import org.springframework.stereotype.Component;
import ru.ioleksiv.telegram.bot.api.annotations.receiver.PreCheckoutQueryReceiver;
import ru.ioleksiv.telegram.bot.api.model.objects.payments.PreCheckoutQuery;
import ru.ioleksiv.telegram.bot.core.controller.handler.unpack.UpdateUnpacker;

import java.util.Optional;

@Component
public class PreCheckoutQueryUnpackerParser extends UnpackerParser<PreCheckoutQueryReceiver, PreCheckoutQuery> {

    @Override
    public Class<PreCheckoutQueryReceiver> getAnnotationClass() {
        return PreCheckoutQueryReceiver.class;
    }

    @Override
    public UpdateUnpacker<PreCheckoutQuery> getUpdateUnpacker() {
        return new UpdateUnpacker<>(in -> Optional.ofNullable(in.getPreCheckoutQuery()), PreCheckoutQuery.class);
    }
}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/core/controller/annotations/parser/unpacker/PreCheckoutQueryUnpackerParser.kt
package io.github.oleksivio.tl.kbot.core.core.controller.annotations.parser.unpacker

import io.github.oleksivio.tl.kbot.core.annotations.receiver.PreCheckoutQueryReceiver
import io.github.oleksivio.tl.kbot.core.core.controller.handler.unpack.UpdateUnpacker
import io.github.oleksivio.tl.kbot.server.api.objects.payments.PreCheckoutQuery
 

class PreCheckoutQueryUnpackerParser : UnpackerParser<PreCheckoutQueryReceiver, PreCheckoutQuery>() {
    override val updateUnpacker: UpdateUnpacker<PreCheckoutQuery>
        get() = UpdateUnpacker(
            { it.preCheckoutQuery },
            PreCheckoutQuery::class
        )

    override val parserAnnotationClass = PreCheckoutQueryReceiver::class
}
